src/GTP.cpp

```diff
@@ -882,10 +894,86 @@ bool GTP::execute(GameState & game, std::string xinput) {
         } else {
             gtp_fail_printf(id, "syntax not understood");
         }
+        return true;
+    } else if (command.find("lz-estimatememory") == 0) {
+        auto base_memory = (size_t)(s_network->get_estimated_size() * Network::CONSERVATIVE_MEMORY_OVERHEAD);
+        auto tree_size = UCTNodePointer::get_tree_size();
+        auto cache_size = s_network->get_estimated_cache_size();
 
+        auto total = base_memory + tree_size + cache_size;
+        printf("Estimated total memory consumption: %d MB.\n", total / 1048576LL);
+        printf("Network with program overhead: %d MB / Search tree: %d MB / %d\n", base_memory / 1048576LL, tree_size / 1048576LL, cache_size / 1048576LL);
         return true;
-    }
+    } else if (command.find("lz-setoption") == 0) {
+        std::istringstream cmdstream(command);
+        std::string tmp, option;
 
+        cmdstream >> tmp >> option;
+
+        if (!cmdstream.fail()) {
```

Comment:
Maybe option handling could be factored out to a separate function, especially as we'll want to add more.